.github/workflows/build_and_test.yml

```diff
@@ -1,11 +1,10 @@
 name: Build and Test
 
-# See https://docs.github.com/en/actions/reference/events-that-trigger-workflows
 on:
-  push:
-    branches:
-      - master
   pull_request:
+    types:
```

Comment:
if we don't specify types it will run these actions on all these events, eating up minutes: https://docs.github.com/en/actions/learn-github-actions/events-that-trigger-workflows#pull_request